ostd/src/irq/bottom_half.rs

```diff
@@ -37,7 +37,7 @@ static BOTTOM_HALF_HANDLER_L2: Once<fn()> = Once::new();
 
 pub(super) fn process() {
     match InterruptLevel::current() {
-        InterruptLevel::L1 => process_l1(),
+        InterruptLevel::L1(_) => process_l1(),
         InterruptLevel::L2 => process_l2(),
         _ => unreachable!("this function must have been call in interrupt context"),
```

Comment:
This is similar to https://github.com/asterinas/asterinas/pull/2429#discussion_r2357495778:
```rust
        InterruptLevel::L1(_) => process_l1(),
        level => {
            debug_assert_eq!(level, InterruptLevel::L2, "this function must have been call in interrupt context");
            process_l2();
        }
```

The last one is in `InterruptLevel::current` (which may be subjective: if you think the bit packing is complicated enough to warrant a release-build `unreachable!()`, I don't have a strong opinion either way; however, debug-build assertions should be clear enough unless our code has bugs):
```rust

            _ => {
                debug_assert_eq!(level, 2, "level must between 0 and 2 (inclusive)");
                Self::L2
            }
```

Reply:
I think this form

```rust
match enum_val {
    VALID_PATTERN_1 => { ... }
    VALID_PATTERN_2 => { ... }
    _ => unreachable!(),
}
```

is in general more clear than

```rust
match enum_val {
    VALID_PATTERN_1 => { ... }
    _ => {
        debug_assert!(enum_val == VALID_PATTERN_2);
    }
}
```

because all branch conditions or patterns are presented symmetrically.

The perceived performance of the latter form is arguably minimal. The fallback branch, `_ => { ... }`, will not be executed (unless the code has a bug). I don't think that we should write code with such micro-optimizations when no evidence is shown that the code is performance critical.